20220831-add-type-restrictions-to-json-syntax.md

```diff
@@ -0,0 +1,558 @@
+# Add Type Restrictions to the JSON Syntax
+
+## Meta
+
+- **Name**: Add Type Restrictions to the JSON Syntax
+- **Start Date**: 2022-08-31
+- **Author(s)**: [rhamzeh](https://github.com/rhamzeh)
+- **Status**: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- **RFC Pull Request**: https://github.com/openfga/rfcs/pull/7
+- **Relevant Issues**:
+  - https://github.com/openfga/rfcs/pull/3
+  - https://github.com/openfga/rfcs/pull/4
+  - https://github.com/openfga/api/pull/27
+- **Supersedes**: N/A
+
+## Summary
+
+Allow users to indicate in their store's authorization model what types of objects can have a particular relation to an object type.
+
+## Definitions
+
+- [OpenFGA JSON Syntax](https://openfga.dev/docs/configuration-language)
+- [OpenFGA DSL](https://openfga.dev/docs/configuration-language)
+- [ListObjects Endpoint](https://openfga.dev/api/service#/Relationship%20Queries/ListObjects)
+- [What is a user?](https://openfga.dev/docs/concepts#what-is-a-user)
+- Floating User ID: A user identifier without a type (e.g. `anne`, `4` or `4179af14-f0c0-4930-88fd-5570c7bf6f59`)
+- Reverse Expansion: Normally [expand](https://openfga.dev/docs/interacting/relationship-queries#expand) takes an object and relation and returns the first leaves of users who are related. Reverse expand would be the opposite, taking a user and a relation and retuning the objects who are related.
+
+## Motivation
+
+### Why should we do this?
```

Comment:
I don't know if it is relevant at this point, but it will also help us implement the indexer (Leopard).